test/e2e/gateway/gatewayapi/gateway_api.go

```diff
@@ -195,6 +195,21 @@ spec:
 			address = net.JoinHostPort(GatewayIP(), "8080")
 		})
 
+		It("should send default static payload for no route", func() {
+			Eventually(func(g Gomega) {
```

Comment:
I think we can use `client.CollectFailure` here

Reply:
no, `client.CollectFailure` which also tries to unmarshal response body to `FailureResponse` which has status code but not the content.

Reply:
Good point, see my other comment